hack/templates/resources/30-cloudprofile.yaml.tpl

```diff
@@ -431,6 +433,89 @@ spec:<% caBundle=value("spec.caBundle", "") %>
         - cn-beijing-f  # Avalibility zone
       % endif
   % endif
+  % if cloud == "packet":
+  packet:
+    constraints:
+      dnsProviders:<% dnsProviders=value("spec.packet.constraints.dnsProviders", []) %>
+      % if dnsProviders != []:
+      ${yaml.dump(dnsProviders, width=10000)}
+      % else:
+      - name: aws-route53
+      - name: unmanaged
+      % endif
+      kubernetes:<% kubernetesVersions=value("spec.packet.constraints.kubernetes.versions", []) %>
+        % if kubernetesVersions != []:
+        ${yaml.dump(kubernetesVersions, width=10000)}
+        % else:
+        versions:
+        - 1.13.3
+        % endif
+      machineImages:<% machineImages=value("spec.packet.constraints.machineImages", []) %>
+      % if machineImages != []:
+      ${yaml.dump(machineImages, width=10000)}
+      % else:
+      - name: "Container Linux - Stable"
```

Comment:
This name is actually the name of the operating system extension. Just FYI

Reply:
"name of the operating system" meaning what?

Reply:
We have OS extension controllers, e.g. https://github.com/gardener/gardener-extensions/tree/master/controllers/os-coreos - this controller reacts on `machineImage.name=coreos`, so in a real Gardener installation on packet you have to use the name of the OS extension here in order to configure the system correctly - in this case `coreos`. For the example it's fine, just wanted to make you  aware.

Reply:
That was the part I didn't fully get. I had thought `name` was just a representative name, or however the consuming cloud provider api treats it. In this case, I was looking more at ID than anything.

So what is in the above-referenced `controllers` dir [here](https://github.com/gardener/gardener-extensions/tree/master/controllers/os-coreos) ? There is only one OS listed (`coreos`). Is that the only thing supported? What is that whole dir for?

Reply:
> For the example it's fine, just wanted to make you aware

Might as well get it correct the first time, if I can.

Reply:
No, `name` is the name for the extension controller (`coreos`). The `id` field you use is cloud provider specific, right?

The OS extension controller is not talking to the cloud provider - it just computes the configuration for the selected operating system. This configuration is passed to MCM which then talks to the API of the cloud (and creates VMs with the generated config). Details: https://github.com/gardener/gardener/blob/master/docs/extensions/operatingsystemconfig.md

Currently, we only have `coreos` support, but generally, the used OS is now pluggable. You can develop an extension for any other OS (@pablochacin is currently developing one for SUSE JeOS).

Reply:
> The id field you use is cloud provider specific, right?

Correct. 

> The OS extension controller is not talking to the cloud provider - it just computes the configuration for the selected operating system

So `name` has actual meaning, then? I had thought it to be a "use what you want" field. Let me summarize then:

* `name` has actual meaning to gardener, should not to the cloud provider, and must match that which is in `controllers`. Right now gardener only supports `coreos`, may support others in the future.
* `id` (and any other fields) has actual meaning to the cloud provider, should have none to gardener. It should be an ID that causes coreos to be deployed onto the machine, so that coreos actually runs, and everything else will work. Eventually, with support for other OSes, there will be other options.

If so, then I should update `name` to be `coreos`. It is just an example, but let's get it correct.

Is all of that correct?

Reply:
> name has actual meaning to gardener, should not to the cloud provider, and must match that which is in controllers. Right now gardener only supports coreos, may support others in the future.

Yes, right. We actually support `coreos` and `coreos-alicloud` (because on Alicloud it's special/different), but true.

> id (and any other fields) has actual meaning to the cloud provider, should have none to gardener. It should be an ID that causes coreos to be deployed onto the machine, so that coreos actually runs, and everything else will work. Eventually, with support for other OSes, there will be other options.

Yes. In the future (with new `Shoot` resource) there will just be `name` and `version`, and then the cloud provider extension has to map this to the correct ID. But for now (with this/old API) it's correct.

> If so, then I should update name to be coreos. It is just an example, but let's get it correct.

Would be better, yes :)

Reply:
> Yes. In the future (with new Shoot resource) there will just be name and version, and then the cloud provider extension has to map this to the correct ID. But for now (with this/old API) it's correct.

Oh, I _really_ like that. I look forward to updating it.

> Would be better, yes :)

Doing now, will update and push out, then resolve this conversation.

Reply:
Done.